main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -3314,6 +3314,10 @@ object Resolver {
 
         case TypeConstructor.Native(clazz) => Result.Ok(clazz)
 
+        case TypeConstructor.MethodReturnType(name, arity) => throw InternalCompilerException("MethodReturnType shouldn't exist.", loc)
+
+        case TypeConstructor.StaticMethodReturnType(clazz, name, arity) => throw InternalCompilerException("StaticMethodReturnType shouldn't exist.", loc)
+
```

Comment:
I suggest to return `Result.Err(ResolutionError.IllegalType(tpe, loc))` in these two cases for now.